source/h1_connection.c

```diff
@@ -518,6 +522,64 @@ struct aws_http_stream *s_new_client_request_stream(const struct aws_http_reques
     return NULL;
 }
 
+static struct h1_stream *s_new_server_stream(struct h1_connection *connection) {
+    struct h1_stream *stream = aws_mem_calloc(connection->base.alloc, 1, sizeof(struct h1_stream));
+    if (!stream) {
+        /*log the bug */
+        return NULL;
+    }
+    int err;
+    { /* BEGIN CRITICAL SECTION */
+        err = aws_mutex_lock(&connection->synced_data.lock);
+        AWS_FATAL_ASSERT(!err);
+        int new_stream_error_code = AWS_ERROR_SUCCESS;
+        if (connection->synced_data.new_stream_error_code) {
```

Comment:
the server stream is always created on the event-loop/channel/connection thread
so we should put the new stream in `thread_data.stream_list`, not `synced_data.pending_stream_list`
and we don't need to acquire the lock to do so